js/layout-javascript/simple-list-code.js

```diff
@@ -2292,99 +2335,126 @@ DynamicList.prototype.sendComment = function(id, value) {
     });
   }
 
-  _this.appendTempComment(id, value, guid, userFromDataSource);
+  var options = {
+    instance: _this,
+    config: _this.data,
+    id: _this.data.id,
+    uuid: _this.data.uuid,
+    container: _this.$container,
+    record: record,
+    comment: value,
+    commentGuid: guid
+  };
 
-  var record = _.find(_this.listItems, { id: id });
+  return Fliplet.Hooks.run('flListDataBeforeNewComment', options).then(function () {
+    value = options.comment;
+    guid = options.commentGuid;
 
-  if (!record) {
-    return Promise.resolve();
-  }
+    if (!value) {
+      return Promise.resolve();
+    }
 
-  if (_.get(record, 'commentCount')) {
-    record.commentCount++;
-  }
+    _this.appendTempComment(id, value, guid, userFromDataSource);
 
-  _this.updateCommentCounter({
-    id: id,
-    record: record
-  });
+    if (_.get(record, 'commentCount')) {
+      record.commentCount++;
```

Comment:
why the `++`?

Reply:
It's incremented by one so that the comment count can be optimistically updated before the request succeeds.